pkg/core/transaction/transaction.go

```diff
@@ -468,3 +469,33 @@ func (t *Transaction) ToStackItem() stackitem.Item {
 		stackitem.NewByteArray(t.Script),
 	})
 }
+
+// Copy creates a deep copy of the Transaction, including all slice fields. Cached values like
+// 'hashed' and 'size' are reset to ensure the copy can be modified independently of the original.
+func (t *Transaction) Copy() *Transaction {
+	if t == nil {
+		return nil
+	}
+	txCopy := *t
+
+	if t.Attributes != nil {
+		txCopy.Attributes = make([]Attribute, len(t.Attributes))
+		copy(txCopy.Attributes, t.Attributes)
```

Comment:
This won't work, we need `Copy()` to be defined on `Attribute` since attribute value is a pointer.

@roman-khimov, agree? Since we're declaring that `Copy()` is a full deep copy.